src/app/admin/modules/data-management/components/data-graph-child/data-graph-child.component.ts

```diff
@@ -0,0 +1,245 @@
+                    position: 'insideTop'
+                }
+            },
+            data: this.seriesData[0].data
+        },
+        {
+            name: '总人数',
+            type: 'bar',
+            stack: '1',
+            label: {
+                normal: {
+                    show: true,
+                    position: 'insideTop'
+                }
+            },
+            data: this.seriesData[1].data
+        }
+    ]
+};
+pieChartOption: EChartOption = {
+
+    title: [{
+        text: this.title,
+        left: 'center',
+        top: 20,
+        textStyle: {
+            color: '#cccc'
+        }
+    }],
+
+    tooltip: {
+        trigger: 'item',
+        formatter: '{a} <br/>{b} : {c} ({d}%)'
+    },
+
+    visualMap: [{
+        show: false,
+        min: 80,
+        max: 600,
+        inRange: {
+            colorLightness: [0, 1]
+        }
+    }],
+    series: [
+        {
+            name: '访问来源',
+            type: 'pie',
+            radius: '55%',
+            center: ['50%', '50%'],
+            data: this.pieGraphData,
+            roseType: 'radius',
+
+            animationType: 'scale',
+            animationEasing: 'elasticOut',
+            animationDelay:  () => {
+                return Math.random() * 200;
+            }
+        }
+    ]
+};
+
+  @Input() set graphParam(val: {
+    selectedGraphType: number,
+    selectedGroupType: number,
+    selectedStartYear: number,
+    selectedEndYear: number,
+    selectedDepartment: string
+  }) {
+    if (!(val && Object.keys(val)))return;
+    this.titleYear = val.selectedStartYear === val.selectedEndYear ?
+        `${val.selectedStartYear}` : `${val.selectedStartYear}-${val.selectedEndYear}`;
+    this.titleDepartment = val.selectedDepartment;
+    this.title = `${this.titleYear} ${this.titleDepartment} ${this.titleGraphNames[val.selectedGraphType].name}`;
+
+    if (this.isPieGraph) {
+        const data: number[] = this.seriesData[0].data;
+        let i: number;
+        this.pieGraphData = [];
+        for (i = 0; i < data.length; i ++) {
+            this.pieGraphData.push({value: data[i], name: this.xAxisList[i]} as PieGraphData);
+        }
+        this.pieGraphData.sort( (a, b) => a.value - b.value);
+        this.pieChartOption.title = [{
+          text: this.title,
+          left: 'center',
+          top: 20,
+          textStyle: {
+              color: '#cccc'
+          }
+        }];
+        this.pieChartOption.series = [
+          {
+              name: '访问来源',
+              type: 'pie',
+              radius: '55%',
+              center: ['50%', '50%'],
+              data: this.pieGraphData,
+              roseType: 'radius',
+              animationType: 'scale',
+              animationEasing: 'elasticOut',
+              animationDelay: () => {
+                  return Math.random() * 200;
+              }
+          }
+        ];
+        echarts.getInstanceByDom(this.myCharts.nativeElement).setOption(this.pieChartOption);
```

Comment:
If you need the reference to the instance of echarts, please refer to doc to replace this line with best practice.